### ISSUE TITLE
Ampache adding duplicates that do not play play (one does)

### ISSUE THREAD

**Issue by Gunrunner101** [ampache/ampache#907]
First let me say how much I love ampache, I use it all the time. I have recently reinstalled ampache and running running 3.8.0.

I have a strange issue, ampache seems to be adding duplicates that are not really there when adding a catalog. I know this because I added a catalog that is a single directory and only has one album in it and no physical duplicates.

In some cases ampache has added the same song 4 times and only one of the 4 actually work. interestingly if it adds one song lets say twice it will do that for the entire album.

I have ran clean and enabled logging and it process everything and removes nothing. Nothing really out of the ordinary as far as I can see.

Does anyone have any suggestions I can try.

Thank you 
Andrew

**Comment by Afterster** [comment]
I cannot reproduce your issue. Please try with latest changes on develop branch on another test installation and see if this is still happening. Also, please send Ampache log file so I can take a look at it. Thanks.

**Comment by krodelabestiole** [comment]
hi
i've got the same issue :
i have some duplicates that appears without cover art and without artist name. the track filenames are the same but they won't play. deleting and recreating the catalog changed nothing, it occures always on the same albums, though i can't find anything weird on mp3 tags. they were mostly processed thru musicbrainz picard.
Here is a part of the log when i try to read such a file. i can't find anything weird either. it's a bit like if the track were 0 sec length.
[ampache.20151117.txt](https://github.com/ampache/ampache/files/36270/ampache.20151117.txt)

**Comment by krodelabestiole** [comment]
here is another log when i play the album that behave correctly
[log_ok.txt](https://github.com/ampache/ampache/files/36275/log_ok.txt)
the only differences i see are the track info (note that the wrong one uses a wrong filename - it should at least return a 404) :

working
> 2015-11-17 06:10:45 [krodelabestiole] (webplayer.class.php) -> Return get_media_js_param {{"title":"Sadhu","artist":"Muslimgauze","artist_id":"6846","album_id":"5541","replaygain_track_gain":"0.000000","replaygain_track_peak":"0.000000","replaygain_album_gain":"0.000000","replaygain_album_peak":"0.000000","media_id":"77534","filetype":"mp3","url":"https:\/\/www.myserver.com\/ampache\/play\/index.php?ssid=987654321098765432109876543210&type=song&oid=77534&uid=3&name=Muslimgauze%20-%20Sadhu.mp3","poster":"https:\/\/www.myserver.com\/ampache\/image.php?object_id=5541&object_type=album&auth=987654321098765432109876543210&thumb=3&name=art.jpg"}}

not working
> 2015-11-17 05:54:43 [krodelabestiole] (webplayer.class.php) -> Return get_media_js_param {{"title":"Sadhu","artist":null,"artist_id":"3642","album_id":"4435","replaygain_track_gain":"0.000000","replaygain_track_peak":"0.000000","replaygain_album_gain":"0.000000","replaygain_album_peak":"0.000000","media_id":"60830","filetype":"mp3","url":"https:\/\/www.myserver.com\/ampache\/play\/index.php?ssid=123456789123456789123456789&type=song&oid=60830&uid=3&name=%20-%20Sadhu.mp3","poster":"https:\/\/www.myserver.com\/ampache\/image.php?object_id=4435&object_type=album&auth=123456789123456789123456789&thumb=3"}} 

and the lack of play and stats event with the wrong one

**Comment by Afterster** [comment]
Ok but as I said :): I cannot reproduce your issue. Please try with latest changes on develop branch on another test installation and see if this is still happening.

**Comment by krodelabestiole** [comment]
i managed to install ampache from the develop branch after some fighting, manually updating composer, importing sql and the user from my old ampache installation because there was none.
but now i have a javascript error : Uncaught SyntaxError: Unexpected token ;
Therefore i cant access ampache settings
Do i really have to try with the develop branch ? i mean if someone had corrected such a bug he would have known it ?

**Comment by krodelabestiole** [comment]
here is the syntax error, index line 139
> var refresh_slideshow_interval=;
i cant find any slideshow time definition in the config file
i changed the slideshow_time value to 10 in the db, (preference table)
it doesnt change anything

**Comment by krodelabestiole** [comment]
i changed line 228 fo templates/header.inc.php to 
> var refresh_slideshow_interval=0<?php echo AmpConfig::get('slideshow_time'); ?>;

(i added a 0) now i can access the settings but the form to add a catalog is not even half loaded, i miss most of the field (i just can name it, no type or path input e.g.)
i think i'm done with the developer branch that just doesnt work to me... unless i did sthg wrong but i cant understand what...

**Comment by krodelabestiole** [comment]
well after some trials i guess i must have been mistaken from the beginning : ampache probably doesn't create duplicates, it just doesn't remove albums when i delete a catalog... the issue is probably related to this. is there something i'm missing (caching functionality or whatever) ? i tried to update everything, check every catalogs, all the buttons i found in the catalogs list but no album will be removed from the music list, even without any catalog left...

**Comment by krodelabestiole** [comment]
sorry then for polluting this thread i guess my issue had nothing to do with this
im creating a new one : https://github.com/ampache/ampache/issues/1013

**Comment by Afterster** [comment]
According to IRC discussions, I believe this is fixed on coming 3.8.1 release (be aware that duplicates will remains if you encountered this issue, but once cleaned it will not appear again).

**Comment by strikerlord** [comment]
I see that the issue has been closed with the last update from Afterster, I do still have a question since my album has a lot of these duplicates as well (even deleting the number from the OS folder and refreshing the album leaves the duplicate ones).

Would you need to recreate the album entirely once release 3.8.1 is out?

**Comment by Afterster** [comment]
I wasn't able to reproduce this issue on latest  changes. Could you provide logs + test files?

**Comment by strikerlord** [comment]
In the attachment below are 4 files;
the log from when the player encountered a duplicated file (explaination of the steps in the top of the file).
screenshot of searching for a song where it gives 4 results while in the local files there is only 1 song and a screenshot of the amount of files it should index, right now it has indexed 21456 songs where it should be around 5000.

also an interesting effect, if i would delete the local file and reindex the songs it will only remove the working song, it will KEEP the duplicates.

[ampache troubleshooting.zip](https://github.com/ampache/ampache/files/92873/ampache.troubleshooting.zip)

**Comment by strikerlord** [comment]
Some new developments;
i deleted the catalog with all the music and after deleting the catelog there are 9611 songs remaining of the previous 21456, but there are not other catalogs.

it seems that some data is not getting cleaned in the database.

**Comment by Afterster** [comment]
@strikerlord try to do it from scratch using web installer by removing ampache.cfg.php

**Comment by strikerlord** [comment]
Alright, i removed the config file and overwritten the existing database while following the web configuration steps.

After this reinstallation it still happens that songs are registered double, but in this case all the duplicates work.
so there is an improvement, but the duplicates still exist.

**Comment by strikerlord** [comment]
For now the issue is "minor", i can handle the amount of duplicates where this doesn't lead to any issues.
the item count is off because of these duplicates (checks out on the total).

Thanks for the help!

**Comment by Afterster** [comment]
@strikerlord what about test files so I can test on my side? Also, please share your settings.

**Comment by strikerlord** [comment]
files and settings are in the attachments;
i have 2 catalogs where I did some searching, the first catalog is located on the System disk (C:) and second on a storage disk (F:).
It seems (random checks) that there are no duplicates indexed from F: but ALL songs on C: are duplicated.

added some ACDC song files which are duplicated, if you need any more let me know.
i could try and see what happens if i remove the ACDC album from C to F, see if the duplicates dissapear, but let me know.

Full config file is also uploaded, marked out the security entries though :).
[ACDC.zip](https://github.com/ampache/ampache/files/110773/ACDC.zip)
[ampache.cfg.zip](https://github.com/ampache/ampache/files/110774/ampache.cfg.zip)

**Comment by Deelight-fr** [comment]
I'm having a similar problem with albums which contain 2 CDs. They appear as two separate entries in the album page (why not) but each of them have their respective tracks duplicated. Both tracks play correctly in my case. I guess something went wrong during the catalog creation (duplicate songs have different but consecutive song_ids).

![soundgarden](https://cloud.githubusercontent.com/assets/15140280/13205605/f710f6e6-d8eb-11e5-930f-e63d4573afad.png)

**Comment by Afterster** [comment]
I tried again with your config, your system and your files without being able to reproduce your issue. I'm afraid you are on your own for the issue identification... Please let us know here when you found out.

**Comment by strikerlord** [comment]
Thank you for the feedback, I will look in to it again once the move is down (got myself a new place :)! ) give me 2 more weeks and i will start working on it again.

**Comment by Phyks** [comment]
@strikerlord Any news on this? Else, if you manage to reproduce from a fresh `develop` branch installation, could you post all the details / config and test files? (Ampache has had quite a bunch of changes since this issue was original posted). Thanks

**Comment by strikerlord** [comment]
I installed one of the development branches (i believe the latest) and started to rebuild the albums with the upload function (which works great now btw!!!).
using this method I managed to get rid of all the duplicates and have not gotten any new duplicates, so this seems to be fixed as of now using this method.
I have not tried to add files through the OS and then updating the albums, so i don't know how that will work.

The songs that used to create duplicates don't do that anymore :).
